src/app/clusters/occupancy-sensor-server/tests/TestOccupancySensingCluster.cpp

```diff
@@ -0,0 +1,1036 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/clusters/occupancy-sensor-server/OccupancySensingCluster.h>
+#include <app/persistence/AttributePersistence.h>
+#include <pw_unit_test/framework.h>
+
+#include <app/clusters/testing/AttributeTesting.h>
+#include <app/clusters/testing/ClusterTester.h>
+#include <app/server-cluster/AttributeListBuilder.h>
+#include <app/server-cluster/testing/TestEventGenerator.h>
+#include <app/server-cluster/testing/TestServerClusterContext.h>
+#include <clusters/OccupancySensing/Attributes.h>
+#include <clusters/OccupancySensing/Metadata.h>
+#include <lib/support/TimerDelegateMock.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::OccupancySensing;
+using namespace chip::Testing;
+
+namespace {
+
+constexpr EndpointId kTestEndpointId = 1;
+
+constexpr OccupancySensing::Structs::HoldTimeLimitsStruct::Type kDefaultHoldTimeLimits = {
+    .holdTimeMin     = 1,
+    .holdTimeMax     = 10,
+    .holdTimeDefault = 1,
+};
+
+constexpr BitMask<OccupancySensing::OccupancyBitmap> kOccupancyUnoccupied = 0;
+
+bool onOccupancyChangedCalled = false;
+bool onHoldTimeChangedCalled  = false;
+
+class TestOccupancySensingDelegate : public OccupancySensingDelegate
+{
+public:
+    void OnOccupancyChanged(bool occupied) override { onOccupancyChangedCalled = true; }
+    void OnHoldTimeChanged(uint16_t holdTime) override { onHoldTimeChangedCalled = true; }
+};
+
+TestOccupancySensingDelegate gTestOccupancySensingDelegate;
+
+struct TestOccupancySensingCluster : public ::testing::Test
+{
+    static void SetUpTestSuite() { ASSERT_EQ(Platform::MemoryInit(), CHIP_NO_ERROR); }
+    static void TearDownTestSuite() { Platform::MemoryShutdown(); }
+
+    void SetUp() override
+    {
+        onOccupancyChangedCalled = false;
+        onHoldTimeChangedCalled  = false;
+    }
+
+    TimerDelegateMock mMockTimerDelegate;
+};
+
+TEST_F(TestOccupancySensingCluster, TestReadClusterRevision)
```

Comment:
The ClusterRevision Logic is not part of OccupancySensingCluster code, I think we might not need this test.